src/app/auth/auth.service.ts

```diff
@@ -34,25 +42,30 @@ export class AuthService {
   }
   logout(): void {
     localStorage.removeItem('jwt_token');
+    this.loggedIn.next(false);
+    this.router.navigate(['/']);
   }
 
   getUserId(): string | null {
     const token = this.getToken();
     if (token) {
       // @ts-ignore
-      const decoded: any = jwt_decode(token);
+      const decoded: any = jwtDecode(token);
       return decoded.userId || null;
     }
     return null;
   }
   isLoggedIn(): boolean {
     const token = this.getToken();
-    return token ? !this.isTokenExpired(token) : false;
+    const isValid =  token ? !this.isTokenExpired(token) : false;
+    this.loggedIn.next(isValid);
```

Comment:
Czemu w tej metodzie wysyłasz next()? To jest w którymś miejscu wywoływana cyklicznie metoda? Czy jak to działa? Wsn czemu tu nie ma po prostu return isValid?

Reply:
to jest model sub/pub wiec po prostu next to dajesz nowy event ktory odbiora wszyscy co subuja wiec wszyscy dostana aktualna wartosc jaka byla pod isValid

Reply:
Wiem jak działa sub/pub. Zastanawia mnie czy nie powinno być na to innej metody, bo wydaje się że ta robi 2 rzeczy na raz

Reply:
No właśnie też trochę nie rozumiem. Rozumiem czemu w login i logout jest next ale tutaj nie jestem pewny. Może po to że w przypadku wygaśnięcia tokenu next nie jest wywoływany (bo nie jest wywoływany login ani logout) i po prostu po sprawdzeniu czy user jest zalogowany (wywołaniu isLoggedIn) powiadamiamy też innych subskrybentów o zmianie stanu zalogowania. Może trzeba jakąś cykliczną metode stworzyć która sprawdze czy token nie wygasł i tam wywoływać next?

Reply:
@Drillllll dobrze się domyśla, login i logout to za mało bo jak token wygaśnie to sie nie zaktualizuje i nie bedziemy mieli autoryzacji a bedzie nam wyswietlalo jakbysmy mieli i dlatego przy sprawdzaniu to dałem, bo w next tasku chcialem zrobic aby w kazdym zadaniu http wywolywana metoda sprawdzala stan i w takim przypadku jak bedziecie mieli komponenty zalezne od tego to wystarczy ze ta zmienna bedziecie subować i bedziecie informowac. Nie wiem jak inaczej zrobic sensownie zeby wlasnie te rzweczy sie komunikowaly niz w ten sposob, jak macie pomysl to dajcie znać